eng/common/scripts/ChangeLog-Operations.ps1

```diff
@@ -40,28 +40,45 @@ function Get-ChangeLogEntriesFromContent {
   $changeLogEntries = [Ordered]@{}
   try {
     # walk the document, finding where the version specifiers are and creating lists
-    $changeLogEntry = $null
     foreach ($line in $changeLogContent) {
       if ($line -match $RELEASE_TITLE_REGEX) {
         $changeLogEntry = [pscustomobject]@{ 
           ReleaseVersion = $matches["version"]
           ReleaseStatus  =  $matches["releaseStatus"]
           ReleaseTitle   = "## {0} {1}" -f $matches["version"], $matches["releaseStatus"]
           ReleaseContent = @()
+          FeaturesAdded = @()
+          BreakingChanges = @()
+          KeyBugsFixed = @()
+          Fixed = @()
         }
         $changeLogEntries[$changeLogEntry.ReleaseVersion] = $changeLogEntry
       }
       else {
         if ($changeLogEntry) {
+          if ($line.Trim() -match "###\s(?<sectionName>.*)")
+          {
+            $sectionName = $matches["sectionName"]
+            $changeLogEntry.ReleaseContent += $line
+            continue
+          }
+
+          if (-not [System.String]::IsNullOrEmpty($line))
+          {
+            switch ($sectionName)
+            {
+              "Features Added" { $changeLogEntry.FeaturesAdded += $line }
```

Comment:
Instead of using a switch statement do you think we should just store the sections into a dictionary so this becomes `$changelogEntry.Sections[$sectionName] += $line`? That way we can potentially any number of sections and just pull them out by name.

Reply:
The only problem here is that the content becomes a string rather than an array of lines.

Reply:
Can't we fix that if we initialize the entry on first use to be an array?